packages/amplify-provider-awscloudformation/src/graphql-resource-manager/amplify-graphql-resource-manager.ts

```diff
@@ -315,7 +315,12 @@ export class GraphQLResourceManager {
         diffs
           // diff.path looks like [ "stacks", "ModelName.json", "Resources", "TableName", "Properties", "KeySchema", 0, "AttributeName"]
           .filter(
-            diff => (diff.kind === 'E' && diff.path.length === 8 && diff.path[5] === 'KeySchema') || diff.path.includes('LocalSecondaryIndexes'),
+            diff => {
+              const keySchemaModified = diff.kind === 'E' && diff.path.length === 8 && diff.path[5] === 'KeySchema';
+              const sortKeyAddedOrRemoved = diff.kind === 'A' && diff.path.length === 6 && diff.path[5] === 'KeySchema' && diff.index === 1;
+              const localSecondaryIndexModified = diff.path.some((pathEntry) => pathEntry === 'LocalSecondaryIndexes');
+              return keySchemaModified || sortKeyAddedOrRemoved || localSecondaryIndexModified;
+            },
```

Comment:
how is this tested?

Reply:
The tests for this are owned in the API category but the functional part is here in the CLI repo, we have identical conditionals for the warning about destructive changes which lives in API, and testing around that. We have an ops task to add testing around this post-release